File: ios/Sources/CapacitorCalendarPlugin/Models/Inputs/DeleteEventInput.swift
import Capacitor
import EventKit

struct DeleteEventInput {
    private let id: String
    private let span: EKSpan

    init(call: CAPPluginCall) throws {
        guard let id = call.getString("id") else {
            throw PluginError.idMissing
        }
        self.id = id
        if let spanInt = call.getInt("span"), let span = EKSpan(rawValue: spanInt) {
            self.span = span
        } else {
            self.span = .thisEvent
        }
    }

    func getId() -> String {
        return id
    }
    
    func getSpan() -> EKSpan {
        return span
    }
}

File: android/src/main/java/dev/barooni/capacitor/calendar/models/inputs/DeleteEventInput.kt
package dev.barooni.capacitor.calendar.models.inputs

import com.getcapacitor.PluginCall
import dev.barooni.capacitor.calendar.PluginError
import dev.barooni.capacitor.calendar.models.enums.EventSpan

data class DeleteEventInput(
    private val call: PluginCall,
) {
    val id = call.getString("id")?.toLongOrNull() ?: throw PluginError.MissingId
    val span: EventSpan = call.getInt("span")?.let { EventSpan.fromInt(it) } ?: EventSpan.THIS_EVENT
}
